src/dir_contents.ml

```diff
@@ -180,6 +180,7 @@ type t =
   ; modules : Modules.t Lazy.t
   ; c_sources : C_sources.t Lazy.t
   ; mlds : (Dune_file.Documentation.t * Path.t list) list Lazy.t
+  ; coq_modules : Coq_module.t list String.Map.t Lazy.t
```

Comment:
Rather than a string map, something like `Coq_module.Name.Map.t` would be preferable. See how `Module.Name.Map.t` works for example.

Reply:
Will do, indeed I was aware of `Module.Name.t` but I thought that maybe it would be overkill for Coq.

Reply:
Umm, not sure I follow here, this is a map from Coq library name to modules, so do you mean I introduce a `Coq_library.t` ? This is on the roadmap, for now Coq library names are just strings.

Reply:
Ah oops. So indeed I misread the code and that the string was a module name :)

> This is on the roadmap, for now Coq library names are just strings.

OCaml library names are just strings as well. But it's still useful to have a separate type for them. It helps with validation, readability, interning. Anyway, as you are the maintainer of the coq plugin, this is only advice. So feel free to ignore it.

Reply:
IMHO it makes a lot of sense to introduce `Coq_library.Name.t`, we will do it later or sooner; maybe I postpone to the PR adding coq library dependencies tho as to keep this PR smaller; the idea here is that we focus on what should be done for the `(include_subdirs qualified)` handling.